mathesar_ui/src/sections/table-view/header/HeaderCell.svelte

```diff
@@ -16,17 +18,29 @@
   export let columnPosition: ColumnPosition;
   export let column: Column;
   export let meta: Meta;
-
+  export let constraintsDataStore: ConstraintsDataStore;
+  
+  let isRequestingToggleAllowDuplicates = false;
+  let menuIsOpen = false;
+  
   $: ({ sort, group } = meta);
   $: sortDirection = ($sort as SortOption)?.get(column.name);
   $: hasGrouping = ($group as GroupOption)?.has(column.name);
 
+  $: allowsDuplicatesStore = constraintsDataStore.columnAllowsDuplicates(column);
```

Comment:
This would create a new store for each component instance of HeaderCell. It would also create new stores whenever the user switches tabs.

We could probably optimize this to avoid creation of multiple stores.